sklearn/compose/_column_transformer.py

```diff
@@ -303,6 +303,10 @@ def set_output(self, *, transform=None):
         for trans in transformers:
             _safe_set_output(trans, transform=transform)
 
+        # Also set output for remainder if it's an estimator
+        if not isinstance(self.remainder, str) and self.remainder not in {"passthrough", "drop"}:
```

Comment:
Nit: I do not think we need to check if `self.remainder` is a string:

```suggestion
        if self.remainder not in {"passthrough", "drop"}:
```